### ISSUE TITLE
[Bug] Improve accessibility for the "Search" button by adding an aria-label

### ISSUE THREAD

**Issue by Silvia-Wachira** [creativecommons/search#290]
## Description
The search button does not have an aria label and that makes it difficult to use because there are some screen-reader users who might not interpret its purpose correctly.

## Reproduction

1.Open the web page.
2. Use the developer tools to review the button element
3. Notice that the button does not have an explicit aria-label

## Expectation
The "Search" button should include an aria-label attribute 
## Screenshots
[screen-capture (6).webm](https://github.com/user-attachments/assets/48458062-b515-48c2-9738-13869df1719d)

## Environment

 - Device: (laptop)
 - OS: (Ubuntu 22.04)
 - Browser: (Chrome)
 - Version: (129.0.6668.100)

## Resolution

- [ x ] I would be interested in resolving this bug.

**Comment by possumbilities** [comment]
The `button` element contains the word `Search`, and in this instance an `aria-label` would be useful to extend or modify the semantics and of the button if it did not have something within its contents indicating its meaning. By default, the contents of a `button` are used as the initial meaning of the button. 

I'm unsure how `search` would necessarily be unclear and need improvement. While not always the case, aria-labels are usually implemented when other semantic options have not occurred or have failed. In this case the primary descriptive text _should_ be descriptive and accessible.

I'm closing for now, but I'm open to clear examples, if there's something I'm missing.
